base/base/shared_ptr_helper.h

```diff
@@ -11,25 +12,33 @@
 template <typename T>
 struct shared_ptr_helper
 {
+private:
+    // https://stackoverflow.com/a/25069711
+    struct make_shared_enabler : T
```

Comment:
```suggestion
    struct make_shared_enabler : public T
```

Reply:
Yep, was thinking it could be related to the failures, but it actually doesn't make sense as `make_shared_enabler` isn't used anywhere directly so visibility specifier shouldn't matter here.
Yeah, let's see how many classes have `final` and are affected by this.